mathics/builtin/tensors.py

```diff
@@ -304,17 +328,37 @@ class Outer(Builtin):
     def eval(self, f, lists, evaluation: Evaluation):
         "Outer[f_, lists__]"
 
+        # If f=!=Times, or lists contain both SparseArray and List, then convert all SparseArrays to Lists
         lists = lists.get_sequence()
         head = None
-        for list in lists:
-            if isinstance(list, Atom):
+        sparse_to_list = f != SymbolTimes
+        contain_sparse = False
+        comtain_list = False
```

Comment:
I believe you `comtain_list` should be `contain_list`

Reply:
Ah, yeah. Fixed it.